tests/test_s3_utils.py

```diff
@@ -167,3 +175,241 @@ def should_raise_any_client_error_other_than_no_such_key(
         s3_url_reader = get_s3_url_reader(get_s3_role_arn(), any_dataset_title(), logger_mock)
         s3_url_reader(any_s3_url())
+
+
+@mark.infrastructure
+def should_return_accurate_etag_value_for_empty_file(s3_client: S3Client) -> None:
+    asset_filename = any_safe_filename()
+    asset_contents = any_file_contents(byte_count=0)
+    expected_empty_file_etag = (
+        '"d41d8cd98f00b204e9800998ecf8427e"'  # we know the etag of empty file
+    )
+
+    staging_bucket = Resource.STAGING_BUCKET_NAME.resource_name
+    storage_bucket = Resource.STORAGE_BUCKET_NAME.resource_name
+
+    local_etag = calculate_s3_etag(asset_contents)
+
+    with S3Object(
+        file_object=BytesIO(initial_bytes=asset_contents),
+        bucket_name=staging_bucket,
+        key=asset_filename,
+    ):
+        try:
+            # import from staging to storage to ensure aws hasn't changed how etag is calculated
+            importer(
+                source_bucket_name=staging_bucket,
+                original_key=asset_filename,
+                target_bucket_name=storage_bucket,
+                new_key=asset_filename,
+                source_s3_client=get_s3_client_for_role(get_s3_role_arn()),
+            )
+
+            s3_etag = get_s3_etag(
+                s3_bucket=storage_bucket, s3_object_key=f"{asset_filename}", logger=get_log()
+            )
+
+            assert (
+                local_etag == expected_empty_file_etag
+            )  # empty file etag should always be the same
+            assert local_etag == s3_etag
+        finally:
+            delete_s3_key(Resource.STORAGE_BUCKET_NAME.resource_name, asset_filename, s3_client)
+
+
+@mark.infrastructure
+def should_return_accurate_etag_value_for_single_chunk_size_file(s3_client: S3Client) -> None:
+    asset_filename = any_safe_filename()
+    asset_contents = any_file_contents(byte_count=1000000)
+
+    staging_bucket = Resource.STAGING_BUCKET_NAME.resource_name
+    storage_bucket = Resource.STORAGE_BUCKET_NAME.resource_name
+
+    local_etag = calculate_s3_etag(asset_contents)
+
+    with S3Object(
+        file_object=BytesIO(initial_bytes=asset_contents),
+        bucket_name=staging_bucket,
+        key=f"{asset_filename}",
+    ):
+        try:
+            # import from staging to storage to ensure aws hasn't changed how etag is calculated
+            importer(
+                source_bucket_name=staging_bucket,
+                original_key=asset_filename,
+                target_bucket_name=storage_bucket,
+                new_key=asset_filename,
+                source_s3_client=get_s3_client_for_role(get_s3_role_arn()),
+            )
+
+            s3_etag = get_s3_etag(
+                s3_bucket=storage_bucket, s3_object_key=f"{asset_filename}", logger=get_log()
+            )
+
+            # ensure single chunk size file
+            assert (
+                len(local_etag) == 34
+            )  # single chunk etag is 32 char, (+2) with quotes as part of etag
```

Comment:
It took me a while to remember this, so it might be good to mention in the comment that this length is specifically only for single-chunk files. You could even avoid the comments by pulling out `34` as something like `SINGLE_CHUNK_ETAG_LENGTH`. and `36` as `MULTI_CHUNK_ETAG_LENGTH`.